src/screens/splash.rs

```diff
@@ -35,48 +40,84 @@ pub(super) fn plugin(app: &mut App) {
         )
             .run_if(in_state(Screen::Splash)),
     );
+
+    // exit the splash screen early if the player hits escape
+    app.add_systems(
+        Update,
+        exit_splash_screen
+            .run_if(input_just_pressed(KeyCode::Escape).and_then(in_state(Screen::Splash))),
+    );
 }
 
 const SPLASH_BACKGROUND_COLOR: Color = Color::srgb(0.157, 0.157, 0.157);
 const SPLASH_DURATION_SECS: f32 = 1.8;
 const SPLASH_FADE_DURATION_SECS: f32 = 0.6;
 
-fn spawn_splash(mut commands: Commands, asset_server: Res<AssetServer>) {
-    commands
-        .ui_root()
-        .insert((
-            Name::new("Splash screen"),
+#[derive(Component)]
+struct SplashScreenContainer;
+
+#[derive(Resource)]
+struct SplashScreenImageList(VecDeque<&'static str>);
```

Comment:
This can be a `Vec` instead of a `VecDeque`.

Reply:
I may be missing something but I feel like `VecDeque` is the better option here as we're popping from the front. 

There is `Vec::pop` which pops from the back, meaning we'd show the images in reverse order. The docs for `pop` actually say:

> If you'd like to pop the first element, consider using VecDeque::pop_front instead.

or there is something like `Vec::drain` but that returns an iterator so there is a lot more boilerplate.

Reply:
That's true. We would either need to store the images in reverse order, use `vec.remove(0)` (when we know the vec is non-empty), or use an index instead of removing the images from the list. The index could reset to 0 when entering `Screen::Splash`.

I normally don't see `VecDeque` unless it's a performance optimization, which isn't relevant in this case.

Reply:
I'll defer to other opinions on this.

Reply:
Yep, happy to change but this is exactly the kind of thing `VecDeque` is in `std::collections` for.

Reply:
Keep it as vec-deque but add a comment about why it's used IMO.